graphstorm-processing/tests/test_dist_heterogenous_loader.py

```diff
@@ -0,0 +1,662 @@
+
+    return data_configs_dict
+
+
+@pytest.fixture(scope="function", name="no_label_data_configs")
+def no_label_data_configs_fixture():
+    """Create data configuration object without labels"""
+    config_path = os.path.join(
+        _ROOT, "resources/small_heterogeneous_graph/gconstruct-no-labels-config.json"
+    )
+
+    with open(config_path, "r", encoding="utf-8") as conf_file:
+        gconstruct_config = json.load(conf_file)
+        gsprocessing_config = GConstructConfigConverter().convert_to_gsprocessing(gconstruct_config)
+
+    data_configs_dict = create_config_objects(gsprocessing_config["graph"])
+
+    return data_configs_dict
+
+
+@pytest.fixture(scope="function", name="dghl_loader")
+def dghl_loader_fixture(spark, data_configs_with_label, tempdir) -> DistHeterogeneousGraphLoader:
+    """Create a re-usable loader that includes labels"""
+    input_path = os.path.join(_ROOT, "resources/small_heterogeneous_graph")
+    dhgl = DistHeterogeneousGraphLoader(
+        spark,
+        local_input_path=input_path,
+        local_output_path=tempdir,
+        output_prefix=tempdir,
+        input_prefix=input_path,
+        data_configs=data_configs_with_label,
+        num_output_files=1,
+        add_reverse_edges=True,
+        enable_assertions=True,
+    )
+    return dhgl
+
+
+@pytest.fixture(scope="function", name="dghl_loader_no_label")
+def dghl_loader_no_label_fixture(
+    spark, no_label_data_configs, tempdir
+) -> DistHeterogeneousGraphLoader:
+    """Create a re-usable loader without labels"""
+    input_path = os.path.join(_ROOT, "resources/small_heterogeneous_graph")
+    dhgl = DistHeterogeneousGraphLoader(
+        spark,
+        local_input_path=input_path,
+        local_output_path=tempdir,
+        output_prefix=tempdir,
+        input_prefix=input_path,
+        data_configs=no_label_data_configs,
+        num_output_files=1,
+        add_reverse_edges=True,
+        enable_assertions=True,
+    )
+    return dhgl
+
+
+@pytest.fixture(scope="function", name="dghl_loader_no_reverse_edges")
+def dghl_loader_no_reverse_edges_fixture(
+    spark, data_configs_with_label, tempdir
+) -> DistHeterogeneousGraphLoader:
+    """Create a re-usable loader that doesn't produce reverse edegs"""
+    input_path = os.path.join(_ROOT, "resources/small_heterogeneous_graph")
+    dhgl = DistHeterogeneousGraphLoader(
+        spark,
+        local_input_path=input_path,
+        local_output_path=tempdir,
+        output_prefix=tempdir,
+        input_prefix=input_path,
+        data_configs=data_configs_with_label,
+        num_output_files=1,
+        add_reverse_edges=False,
+        enable_assertions=True,
+    )
+    return dhgl
+
+
+@pytest.fixture(scope="session", name="user_rated_movie_df")
+def user_rated_movie_df_fixture(spark: SparkSession) -> DataFrame:
+    """User rated movie edges DataFrame"""
+    data_path = os.path.join(
+        _ROOT, "resources/small_heterogeneous_graph/edges/user-rated-movie.csv"
```

Comment:
How do you create resources/small_heterogeneous_graph/edges/user-rated-movie.csv?
Can we use some script to create it?

Reply:
I'd rather use the static file than write code to re-create it every time. Unless we have good reason to use code to create these files, I'd prefer to keep the static files for now at least.

Reply:
So the test data will be stored in our CI image?

Reply:
I'm storing the test data in the repo itself, it's around 200KB of data.

Reply:
We'll solve this in another PR, resolving here as the test files are not part of this PR.